### PR TITLE
Documentation snapshot v3.9.0

### PR DESCRIPTION
**Pull request by davidcok**
Strict copy of the documentation (docs folder) at the time of 3.9.0, with the exception of the added link in Snapshots.md and setting the version numbers

### CODE REVIEW

File: latest/Snapshots.md [hunk 1]
```diff
@@ -6,4 +6,8 @@ layout: default
 ## Dafny Documentation snapshots
 
 - [Current development version](https://dafny.org/dafny)
-- [Latest release snapshot](https://dafny.org)
+- [Latest release snapshot](https://dafny.org/latest)
+- [v3.9.0](https://davidcok.github.io/dafny-lang.github.io/v3.9.0)
+- [v3.9.0](https://davidcok.github.io/dafny-lang.github.io/v3.9.0)
+- [v3.9.0](https://davidcok.github.io/dafny-lang.github.io/v3.9.0)
+- [v3.9.0](https://davidcok.github.io/dafny-lang.github.io/v3.9.0)
```
Comment (cpitclaudel) [inline latest/Snapshots.md:13]: Multiple links again

File: latest/DafnyRef/version.txt [hunk 2]
```diff
@@ -1 +0,0 @@
-Latest release snapshot
```
Comment (cpitclaudel) [inline latest/DafnyRef/version.txt:1]: Why is this file gone?

**Review by cpitclaudel — CHANGES_REQUESTED** [review]
I can't actually check all the new files, so I trust you that they match.  I spotted a few typos.

File: latest/Snapshots.md (hunk 1 above)
Comment (davidcok) [reply latest/Snapshots.md:13]: Aargh - should have found this before files were copies.  Fixed.

File: latest/DafnyRef/version.txt (hunk 2 above)
Comment (davidcok) [reply latest/DafnyRef/version.txt:1]: Replaced - not sure how it disappeared.

**Review by cpitclaudel — APPROVED** [review]

**Pull request merged**